build/Common.props

```diff
@@ -28,18 +28,18 @@
     <NetFrameworkSupportedVersions>net481;net48;net472;net471;net47;net462</NetFrameworkSupportedVersions>
 
     <!-- production TFMs -->
-    <TargetFrameworksForLibraries>net8.0;net6.0;netstandard2.0;$(NetFrameworkMinimumSupportedVersion)</TargetFrameworksForLibraries>
-    <TargetFrameworksForLibrariesExtended>net8.0;net6.0;netstandard2.1;netstandard2.0;$(NetFrameworkMinimumSupportedVersion)</TargetFrameworksForLibrariesExtended>
-    <TargetFrameworksForPrometheusAspNetCore>net8.0;net6.0</TargetFrameworksForPrometheusAspNetCore>
+    <TargetFrameworksForLibraries>net9.0;net8.0;netstandard2.0;$(NetFrameworkMinimumSupportedVersion)</TargetFrameworksForLibraries>
+    <TargetFrameworksForLibrariesExtended>net9.0;net8.0;netstandard2.1;netstandard2.0;$(NetFrameworkMinimumSupportedVersion)</TargetFrameworksForLibrariesExtended>
+    <TargetFrameworksForPrometheusAspNetCore>net9.0;net8.0</TargetFrameworksForPrometheusAspNetCore>
```

Comment:
One more thing, should we really add there net9? .NET8 will be supported longer than .NET8.
Previously there were no packages targeted .NET7. Ref: #5795

Reply:
@CodeBlanch What are your thoughts on it?

Reply:
I don't see any particular advantage with a net9 target...

Reply:
Reminds me of this: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4918

Question being, why did we add `net8.0` to everything? 🤣 At the time I tried to block that because it exploded the amount of public API files we had and maintaining those is a pain. But I went and changed how the public API stuff works so we don't need public API files for a TFM unless there aren't customizations for that TFM.

Where we are now I don't have an issue with adding `net9.0`. We did it for `net8.0`. Which I guess is to say I'm fine with each year adding the latest version as a target when we upgrade. Don't have a strong reason for that other than when users look at NuGet and they see explicit targets I guess it is strong indication we were intentional 🤷

/cc @alanwest

Reply:
Was `net8.0` more valid to add because it's a LTS release? `net7.0` end of support is before `net6.0`, so it'll be removed before `net6.0` anyways. `net8.0`, however, will outlive `net6.0`. Not adding `net8.0` will unavoidably leave a gap somewhere in the timeline.

![image](https://github.com/user-attachments/assets/be8bfa2f-b7a5-4099-b9b8-928d26b38c86)

Reply:
> I don't see any particular advantage with a net9 target...

Me either, and as @CodeBlanch points out this is something we have pushed against before, and honestly I never understood the reasons for adding a bunch of targets.

It stems back to this comment https://github.com/open-telemetry/opentelemetry-dotnet/issues/4591#issuecomment-1591854918. But again, I never understood the benefits to the OpenTelemetry .NET project.

All that said, I'm fine just adding net9.0 everywhere and continuing to make that the pattern going forward. If we want to circle back to this decision then I'd prefer it be in the context of a conversation about removing _all_ unnecessary targets and going back to the state we had before.